### DISCUSSION TITLE
Prevent or Remove Deanonymization

### DISCUSSION THREAD

**Discussion by hadupa** [microsoft/presidio#1371] [Q&A]
Hey Everyone,

I am running a local instance of Presidio and was curious if there is a way to prevent Deanonymization altogether? Not only for the user but to prevent Presidio from storing anonymized PII Entities "in the backend" as the documentation describes. If not, where is that "in the backend" and can it be configured for easy removal or be cleared after each use?

Thanks!

**Comment by omri374** [comment]
Hi, could you please elaborate on what you mean by preventing deanonymization?
Presidio does not store any PII at any point.

**Reply by hadupa** [reply]
Forgive me, I am looking for where I read that in the documentation and can't find it promptly. I will keep looking in the meantime and provide a link once I've found it. 

In the interim, if you know any ways to prevent Deanonymization I would love to hear about them. My concern is that, once a text is Anonymized, someone may be able to use Presidio or some other service to Deanonymize the text. If my concern is baseless then I apologize and thank you for your help.

**Reply by omri374** [reply]
In a nutshell, to prevent any sort of re-identification, one would have to remove all direct and indirect identifiers. Presidio focuses on direct identifiers, and cannot guarantee a complete anonymization.
